src/SA/probe/entry.c

```diff
@@ -0,0 +1,73 @@
+#include "bofdefs.h"
+#include "beacon.h"
+#include "base.c"
+
+BOOL is_port_open(char* host, int port)
+{
+    BOOL ret = FALSE;
+    struct addrinfo* result = NULL;
+    struct addrinfo hints;
+
+    intZeroMemory(&hints, sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_protocol = IPPROTO_TCP;
+
+    char port_str[8] = { 0 };
+    MSVCRT$sprintf(port_str, "%d", port);
+
+    if (WS2_32$getaddrinfo(host, port_str, &hints, &result) == ERROR_SUCCESS) {
+        SOCKET sock = WS2_32$socket(result->ai_family, result->ai_socktype, result->ai_protocol);
+        if (sock) {
+            u_long nonblock = 1;
+            WS2_32$ioctlsocket(sock, FIONBIO, &nonblock);
+
+            struct timeval tv;
+            tv.tv_sec = 5;
+            tv.tv_usec = 0;
+
+            struct fd_set sockets;
+            FD_ZERO(&sockets);
+            FD_SET(sock, &sockets);
+
+            WS2_32$connect(sock, result->ai_addr, result->ai_addrlen);
+            WS2_32$select(1, NULL, &sockets, NULL, &tv);
+
+            if (WS2_32$__WSAFDIsSet(sock, &sockets)) {
+                WS2_32$closesocket(sock);
```

Comment:
from (https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-closesocket)[https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-closesocket]
`An application should always have a matching call to closesocket for each successful call to socket to return any socket resources to the system.`

from that we should move this closesocket call below this if statement, rather then inside it